src/history/history.go

```diff
@@ -28,6 +28,13 @@ func initState() error {
 	if err := json.Unmarshal(bytes, &history); err != nil {
 		return fmt.Errorf("cannot unmarshal json to history - %v", err)
 	}
+
+	for repoURL, service := range history {
+		if service.StateTag == "" {
+			service.StateTag = makeTag(repoURL, *service.Current)
+		}
```

Comment:
Also check if the tag is invalid or not. That is the stored tag is the same as what should be there. If it is different, log a warning and write the correct tag.